File: ruby/roman_numerals/spec/roman_numerals_spec.rb
require 'spec_helper'
require 'fixnum'

describe "decimal to roman" do
  it "converts decimal to roman" do
    expect(1.to_roman).to eq("I")
    expect(2.to_roman).to eq("II")
    expect(3.to_roman).to eq("III")
    expect(5.to_roman).to eq("V")
    expect(7.to_roman).to eq("VII")
    expect(10.to_roman).to eq("X")
    expect(18.to_roman).to eq("XVIII")
    expect(30.to_roman).to eq("XXX")
    expect(4.to_roman).to eq("IV")
    expect(9.to_roman).to eq("IX")
    
    expect(1984.to_roman).to eq("MCMLXXXIV")
    expect(3999.to_roman).to eq("MMMCMXCIX")
  end
end

File: typescript/roman_numerals/spec/roman_numerals.spec.ts
import {roman_of} from "../src/roman";

describe("Roman of", () => {
    [
        {arabic: 0, roman: '' },
        {arabic: -3, roman: '' },
        {arabic: 1, roman: 'I' },
        {arabic: 2, roman: 'II' },
        {arabic: 3, roman: 'III' },
        {arabic: 5, roman: 'V' },
        {arabic: 6, roman: 'VI' },
        {arabic: 7, roman: 'VII' },
        {arabic: 8, roman: 'VIII' },
        {arabic: 10, roman: 'X' },
        {arabic: 17, roman: 'XVII' },
        {arabic: 20, roman: 'XX' },
        {arabic: 4, roman: 'IV' },
        {arabic: 9, roman: 'IX' },
        {arabic: 40, roman: 'XL' },
        {arabic: 50, roman: 'L' },
        {arabic: 90, roman: 'XC' },
        {arabic: 100, roman: 'C' },
        {arabic: 400, roman: 'CD' },
        {arabic: 500, roman: 'D' },
        {arabic: 1000, roman: 'M' },
        {arabic: 3888, roman: 'MMMDCCCLXXXVIII' },
    ].forEach(data => {
        it(`${data.arabic} should be ${data.roman}`, () => {
            expect(roman_of(data.arabic)).toBe(data.roman);
        })});
});
